syntax = "proto3";

package flowpipe.stages.csv.arrow.sink.v1;

// Configuration for CsvArrowSink stage
message CsvArrowSinkConfig {
  enum FileSystem {
    FILE_SYSTEM_AUTO = 0;
    FILE_SYSTEM_LOCAL = 1;
    FILE_SYSTEM_S3 = 2;
    FILE_SYSTEM_GCS = 3;
    FILE_SYSTEM_HDFS = 4;
  }

  enum Compression {
    COMPRESSION_AUTO = 0;
    COMPRESSION_UNCOMPRESSED = 1;
    COMPRESSION_SNAPPY = 2;
    COMPRESSION_GZIP = 3;
    COMPRESSION_BROTLI = 4;
    COMPRESSION_ZSTD = 5;
    COMPRESSION_LZ4 = 6;
    COMPRESSION_LZ4_FRAME = 7;
    COMPRESSION_LZO = 8;
    COMPRESSION_BZ2 = 9;
  }

  enum QuotingStyle {
    QUOTING_STYLE_NEEDED = 0;
    QUOTING_STYLE_ALL_VALID = 1;
    QUOTING_STYLE_NONE = 2;
  }

  // File path to write
  string path = 1;

  // Append to existing file instead of truncating
  bool append = 2;

  // WriteOptions
  optional bool include_header = 3;
  string delimiter = 4;
  optional QuotingStyle quoting_style = 5;
  optional string null_string = 6;

  // Filesystem selection
  FileSystem filesystem = 7;

  // Compression selection
  Compression compression = 8;
}
